lms/djangoapps/branding/serializers.py

```diff
@@ -0,0 +1,40 @@
+"""
+Serializers for the branding app.
+"""
+
+from rest_framework import serializers
+
+import lms.djangoapps.branding.toggles as branding_toggles
+
+
+class WaffleFlagsSerializer(serializers.Serializer):
+    """
+    Serializer for waffle flags.
+    """
+    use_new_index_page = serializers.SerializerMethodField()
+    use_new_catalog_page = serializers.SerializerMethodField()
+    use_new_course_about_page = serializers.SerializerMethodField()
+
+    def get_course_key(self):
+        """
+        Retrieve the course_key from the context.
+        """
+        return self.context.get("course_key")
+
+    def get_use_new_index_page(self, obj):
+        """
+        Returns whether the new index page is enabled.
+        """
+        return branding_toggles.use_new_index_page()
```

Comment:
In order to obtain the Waffle Flag value for a specific course, you must pass its course ID as an argument.